streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamAggregate.java

```diff
@@ -118,10 +120,13 @@ public void process(final Record<KIn, VIn> record) {
 
             newAgg = aggregator.apply(record.key(), record.value(), oldAgg);
 
-            store.put(record.key(), newAgg, newTimestamp);
-            tupleForwarder.maybeForward(
-                record.withValue(new Change<>(newAgg, sendOldValues ? oldAgg : null))
-                    .withTimestamp(newTimestamp));
+            final long putReturnCode = store.put(record.key(), newAgg, newTimestamp);
```

Comment:
Given that we compute `newTimestamp` as max, it should only move forward, even for out-of-order records, so I think this processor does not need an update?

It actually also implies, that we don't really "fix the history" for this case either (what I think is fine), but we can simplify this code (we might also want to call this out on the KIP?)

Reply:
Yes, we do not attempt to "fix" older aggregation results. I can add a note in the KIP. 

See my comment above about whether we can actually guarantee that result timestamps are nondecreasing: https://github.com/apache/kafka/pull/13564#discussion_r1171925337 Same question applies for both table aggregations and stream aggregations.